bin/ChakraCore/ChakraCoreShared.cpp

```diff
@@ -0,0 +1,33 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "Runtime.h"
+#include "Core/ConfigParser.h"
+#include "JsrtContext.h"
+#include "TestHooks.h"
+
+#ifdef __APPLE__
+// dummy usage of JSRT to force export JSRT on dylib
+#include "ChakraCore.h"
+void DummyJSRTCall()
+{
+    JsRuntimeHandle *runtime;
+    JsRuntimeAttributes attr;
+    JsCreateRuntime(attr, nullptr, runtime);
+}
+#endif
+
+__attribute__ ((visibility ("default")))
+EXTERN_C BOOL WINAPI DllMain(HINSTANCE hmod, DWORD dwReason, PVOID pvReserved)
+{
+    // xplat doesn't need DllMain is being called.
+    // Instead, we handle attach / detach stuff internally
```

Comment:
Okay. Added the location of the file that is responsible from the internal mechanism. There are additional comments there.